packages/@atjson/offset-annotations/src/annotations/reddit-embed.ts

```diff
@@ -0,0 +1,51 @@
+import { BlockAnnotation } from "@atjson/document";
+
+/**
+ * Reddit post embeds
+ */
+export class RedditEmbed extends BlockAnnotation<{
+  /**
+   * The embed URL
+   */
+  url: string;
+
+  /**
+   * Whether the post's content should be hidden
+   */
+  hidePostContent?: boolean;
+
+  /**
+   * The post's content will be hidden if it has been
+   * edited after embedding.
+   */
+  hidePostContentIfEditedAfter?: string;
+
+  /**
+   * Whether the username should be blurred.
+   */
+  hideUsername?: boolean;
+
+  /**
+   * The height of the embed
+   */
+  height: number;
+
+  /**
+   * Post contents to store prior to load.
+   */
+  content: string;
+
+  /**
+   * Layout information, used to indicate mutually
+   * exclusive layouts, for example sizes, floats, etc.
+   */
+  layout?: string;
+
+  /**
+   * A named identifier used to quickly jump to this item
+   */
+  anchorName?: string;
+}> {
+  static type = "reddit-embed";
+  static vendorPrefix = "offset";
+}
```

Comment:
Reddit embeds also have an option to display in dark mode, do we want to support that option?

```
<blockquote class="reddit-embed-bq" style="height:500px" data-embed-theme="dark" data-embed-height="740"><a href="https://www.reddit.com/r/DIY/comments/1ddx4kp/whats_going_on_with_my_washer_discharge_hose_and/">[What's going on with my washer discharge hose and how can I prevent it in the future?](https://www.reddit.com/r/DIY/comments/1ddx4kp/whats_going_on_with_my_washer_discharge_hose_and/)</a><br> by<a href="https://www.reddit.com/user/Tsiah16/">[u/Tsiah16](https://www.reddit.com/user/Tsiah16/)</a> in<a href="https://www.reddit.com/r/[DIY](https://www.reddit.com/r/DIY/)/">DIY</a></blockquote><script async="" src="https://embed.reddit.com/widgets.js" charset="UTF-8"></script>
```

Reply:
I think that is something that is up to the calling context and not necessarily something that should be controlled by editors.